core/modules/memman/MemFileSet.h

```diff
@@ -76,27 +78,57 @@ class MemFileSet {
     //! @bried Lock all of the required tables in a table set and as many
     //!        flexible files as possible.
     //!
+    //! @param strict- When true, if a required table could not be locked, its
+    //!                memory remains reserved but otherwise locking continues.
+    //!                When false, execution stops when an error is encountered.
+    //!
     //! @return =0 all required bytes that could be locked were locked.
     //! @return !0 A required file could not be locked, errno value is returned.
     //-----------------------------------------------------------------------------
 
-    int    lockAll();
+    int    lockAll(bool strict);
 
     //-----------------------------------------------------------------------------
-    //! @brief Return status.
+    //! @bried Map all of the required tables in a table set and as many
+    //!        flexible files as possible.
     //!
-    //! @return Status information.
+    //! @return =0 all required tables that could be mapped were mapped.
+    //! @return !0 A required file could not be mapped, errno value is returned.
     //-----------------------------------------------------------------------------
 
-    MemMan::Status status();
+    int    mapAll();
+
+    //-----------------------------------------------------------------------------
+    //! @brief Control serial access to this object. When obtaining a lock,
+    //!        this method should be called with a common mutex locked.
+    //!
+    //! @param dolok If true, obtains a mutex on this object to serialize access.
+    //!              Otherwise, the lock is released.
+    //!
+    //! @return true  Normal return.
+    //! @return false lktst was true and serialization mutex was already locked.
+    //-----------------------------------------------------------------------------
+
+    bool serialize(bool dolok) {
```

Comment:
The code can be re-written using shared pointers and lock guards so this function isn't needed.